torch/onnx/_internal/exporter/_compat.py

```diff
@@ -197,6 +197,11 @@ def export_compat(
                 keep_initializers_as_inputs=keep_initializers_as_inputs,
             )
             onnx_program = _onnx_program.ONNXProgram(ir.load(f), None)
+
+            # NOTE: It it's falling back to the legacy exporter, we don't need to
+            # optimize the model, so we return it here. Users can still optimize
+            # the model using the optimize() if they want.
+            return onnx_program
```

Comment:
Do we still want to call convert_version? Or no?

Reply:
Good point! We should avoid that as well. I will update this.

Reply:
Changed the hard coded 17 to opset version

Reply:
If a user sets opset 18, it is going to fail. Do we want that? (Right now version conversion doesn't do anything)

Reply:
Could you explain why does it fail on 18? I believe we have symbolic_opset18.py. Or do you mean we should return a stable/known supported opset version, in case users set it to something even torchscript exporter is not supporting? If it's later, I think it is fine to let user have opset version is not supported error message, because opset_version is what they povide(want).

Reply:
I believe that the support for symbolic_opset18 is incomplete (specifically, reduce ops are not handled)?

Reply:
Reverted

Reply:
Users would probably put a higher opset version when they use torch.onnx.export, so here is likely to get something torchscript exporter does not support. I am reverting it back to hardcoded just in case.